src/utils/compact.ts

```diff
@@ -0,0 +1,17 @@
+type FalseyVals = '' | 0 | false | null | undefined;
+
+const notFalsey = <T>(
+   value: T | FalseyVals
+): value is T => {
+   return (
+      value !== '' &&
+      value !== 0 &&
+      value !== false &&
+      value !== null &&
+      value !== undefined
+   );
+};
```

Comment:
Minor formatting nit.

```suggestion
function notFalsy<T>(value: T | FalsyVals): value is T {
   return value !== '' 
      && value !== 0
      && value !== false
      && value !== null
      && value !== undefined;
};
```